pkg/build/api/types.go

```diff
@@ -65,19 +75,88 @@ type BuildSpec struct {
+	// Compute resource requirements to execute the build.
 	Resources kapi.ResourceRequirements
 
 	// PostCommit is a build hook executed after the build output image is
 	// committed, before it is pushed to a registry.
 	PostCommit BuildPostCommitSpec
 
-	// Optional duration in seconds, counted from the time when a build pod gets
-	// scheduled in the system, that the build may be active on a node before the
-	// system actively tries to terminate the build; value must be positive integer
+	// Optional duration in seconds, counted from the time when a build pod
+	// gets scheduled in the system, that the build may be active on a node
+	// before the system actively tries to terminate the build; value must be
+	// positive integer.
 	CompletionDeadlineSeconds *int64
 }
 
+// BuildTriggerCause holds information about the trigger of a build.
+type BuildTriggerCause struct {
+	// Reason is used to store a human readable reason for why the build was
+	// triggered. E.g.: "Build manually triggered by user", "Build triggered by
+	// ConfigChange update", "Build triggered by ImageChange for image:
+	// FOO:BAR", etc.
+	// TODO: Maybe this just be reworded to hold short "codes", while the human
+	// readable text is computed.
+	Reason BuildTriggerReason
+
+	// CIDetails holds information about the CI server job that triggered a
+	// specific build.
+	CIDetails *CIJobDetails
+
+	// GitHubWebHook represents data for a GitHub webhook that fired a specific
+	// build.
+	GitHubWebHook *GitHubWebHookInfo
+
+	// GenericWebHook holds data about a builds generic webhook trigger.
+	GenericWebHook *GenericWebHookTriggerInfo
+
+	// ImageChangeBuild stores information about an imagechange event that
+	// triggered a new build.
+	ImageChangeBuild *ImageChangeBuildInfo
+}
+
+// BuildTriggerReason stores a human readable message about the cause of a
+// triggered build.
+type BuildTriggerReason string
+
+// CIJobDetails holds information about specific CI server jobs.
+type CIJobDetails struct {
+	JenkinsDetails JenkinsJobDetails
+}
+
+// JenkinsJobDetails holds information about a Jenkins job.
+type JenkinsJobDetails struct {
+	// JobURL stores a link that points to a Jenkins job.
+	JobURL string // REVIEW: Maybe this should be a *url.URL.
+}
+
+// GenericWebHookTriggerInfo holds information about a generic WebHook that
+// triggered a build
+type GenericWebHookTriggerInfo string
+
+// GitHubWebHookInfo has information about a GitHub webhook that triggered a
+// build
+type GitHubWebHookInfo struct {
+	// Source is the git source of the github webhook that triggered the
+	// build
+	Source GitBuildSource
+
+	// Revision is the git revision of the github webhook that triggered the
+	// build
+	Revision GitSourceRevision
+}
+
+// ImageChangeBuildInfo has information about the last imagechange trigger that
+// fired a new build
+type ImageChangeBuildInfo struct {
+	// ImageID is the image id that triggered the build
+	ImageID string
```

Comment:
is this really an ID or just "name", I thought we dropped 'ID' from API... @smarterclayton ?